// ng-uikit-pro-standard/esm2020/lib/pro/auto-format/mdb-cvv.directive.mjs
import { Directive, HostListener, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class MdbCvvDirective {
    constructor() {
        this.maxLength = '4';
    }
    onInput(event) {
        this.formatInput(event);
    }
    formatInput(event) {
        const input = event.target.value;
        const newValue = this.getFormattedValue(input);
        event.target.value = newValue;
    }
    getFormattedValue(value) {
        value = this.removeNonDigits(value);
        return value;
    }
    removeNonDigits(value) {
        return value.replace(/\D/g, '');
    }
}
MdbCvvDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbCvvDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MdbCvvDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.7", type: MdbCvvDirective, selector: "[mdbCvv]", host: { listeners: { "input": "onInput($event)" }, properties: { "attr.maxLength": "this.maxLength" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbCvvDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[mdbCvv]',
                }]
        }], propDecorators: { maxLength: [{
                type: HostBinding,
                args: ['attr.maxLength']
            }], onInput: [{
                type: HostListener,
                args: ['input', ['$event']]
            }] } });
