// ng-uikit-pro-standard/esm2020/lib/pro/accordion/index.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SqueezeBoxComponent } from './components/squeezebox';
import { SBItemComponent } from './components/sb-item';
import { SBItemHeadComponent } from './components/sb-item.head';
import { SBItemBodyComponent } from './components/sb-item.body';
import * as i0 from "@angular/core";
export const SQUEEZEBOX_COMPONENTS = [SqueezeBoxComponent, SBItemComponent, SBItemHeadComponent, SBItemBodyComponent];
export { SBItemComponent } from './components/sb-item';
export { SBItemHeadComponent } from './components/sb-item.head';
export { SBItemBodyComponent } from './components/sb-item.body';
export { SqueezeBoxComponent } from './components/squeezebox';
export class AccordionModule {
}
AccordionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AccordionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccordionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AccordionModule, declarations: [SqueezeBoxComponent, SBItemComponent, SBItemHeadComponent, SBItemBodyComponent], imports: [CommonModule], exports: [SqueezeBoxComponent, SBItemComponent, SBItemHeadComponent, SBItemBodyComponent] });
AccordionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AccordionModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AccordionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [SQUEEZEBOX_COMPONENTS],
                    exports: [SQUEEZEBOX_COMPONENTS]
                }]
        }] });
