// ng-uikit-pro-standard/esm2020/lib/pro/accordion/components/sb-item.mjs
import { Component, ContentChild, Input, } from '@angular/core';
import { SBItemBodyComponent } from './sb-item.body';
import { coerceBooleanProperty } from '@angular/cdk/coercion';
import * as i0 from "@angular/core";
import * as i1 from "../mdb-accordion.service";
import * as i2 from "@angular/common";
export class SBItemComponent {
    constructor(accordionService, _cdRef) {
        this.accordionService = accordionService;
        this._cdRef = _cdRef;
        this._collapsed = true;
        this.autoExpand = true;
        this.idModifier = Math.floor(Math.random() * 1000);
    }
    get collapsed() {
        return this._collapsed;
    }
    set collapsed(value) {
        this._collapsed = coerceBooleanProperty(value);
    }
    ngAfterViewInit() {
        if (this.body !== undefined) {
            setTimeout(() => {
                this.collapsed
                    ? (this.body.expandAnimationState = 'collapsed')
                    : (this.body.expandAnimationState = 'expanded');
            }, 0);
            this.body.toggle(this.collapsed);
            if (this.autoExpand !== false) {
                this.body.openSidenavOnActiveLink();
            }
        }
        if (this.body) {
            this.body.autoExpand = this.autoExpand;
            this.body.collapsed = this.collapsed;
        }
    }
    ngAfterContentInit() {
        setTimeout(() => {
            if (this.body && this.body.expandAnimationState === 'expanded') {
                this.collapsed = false;
            }
        }, 40);
        if (this.body) {
            this.body.id = `mdb-accordion-body-${this.idModifier}`;
        }
    }
    toggle(collapsed) {
        this.accordionService.didItemToggled(this);
        this.applyToggle(collapsed);
    }
    applyToggle(collapsed) {
        if (this.body !== undefined) {
            this.collapsed = collapsed;
            this.body.toggle(collapsed);
            this._cdRef.markForCheck();
        }
    }
}
SBItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SBItemComponent, deps: [{ token: i1.MdbAccordionService }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
SBItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.2.7", type: SBItemComponent, selector: "mdb-item, mdb-accordion-item", inputs: { collapsed: "collapsed", customClass: "customClass" }, queries: [{ propertyName: "body", first: true, predicate: SBItemBodyComponent, descendants: true }], exportAs: ["sbItem"], ngImport: i0, template: "<div class=\"card {{ customClass }}\" [ngClass]=\"{'is-collapsed': collapsed, 'active': !collapsed}\">\n  <ng-content></ng-content>\n</div>", directives: [{ type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SBItemComponent, decorators: [{
            type: Component,
            args: [{ exportAs: 'sbItem', selector: 'mdb-item, mdb-accordion-item', template: "<div class=\"card {{ customClass }}\" [ngClass]=\"{'is-collapsed': collapsed, 'active': !collapsed}\">\n  <ng-content></ng-content>\n</div>" }]
        }], ctorParameters: function () { return [{ type: i1.MdbAccordionService }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { collapsed: [{
                type: Input
            }], customClass: [{
                type: Input
            }], body: [{
                type: ContentChild,
                args: [SBItemBodyComponent]
            }] } });
