// ng-uikit-pro-standard/esm2020/lib/pro/select/select.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbSelectComponent } from './select.component';
import { MdbOptionModule } from './../option/option.module';
import { OverlayModule } from '@angular/cdk/overlay';
import { MdbSelectFilterComponent } from './select-filter.component';
import * as i0 from "@angular/core";
export class MdbSelectModule {
}
MdbSelectModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdbSelectModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbSelectModule, declarations: [MdbSelectComponent, MdbSelectFilterComponent], imports: [CommonModule, MdbOptionModule, OverlayModule], exports: [MdbSelectComponent, MdbSelectFilterComponent, MdbOptionModule] });
MdbSelectModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbSelectModule, imports: [[CommonModule, MdbOptionModule, OverlayModule], MdbOptionModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MdbSelectComponent, MdbSelectFilterComponent],
                    imports: [CommonModule, MdbOptionModule, OverlayModule],
                    exports: [MdbSelectComponent, MdbSelectFilterComponent, MdbOptionModule],
                }]
        }] });
