// ng-uikit-pro-standard/esm2020/lib/free/tables/directives/mdb-table-row.directive.mjs
import { Directive, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
export class MdbTableRowDirective {
    constructor(el) {
        this.el = el;
        this.rowCreated = new EventEmitter();
        this.rowRemoved = new EventEmitter();
    }
    ngOnInit() {
        this.rowCreated.emit({ created: true, el: this.el.nativeElement });
    }
    ngOnDestroy() {
        this.rowRemoved.emit({ removed: true });
    }
}
MdbTableRowDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTableRowDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
MdbTableRowDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.7", type: MdbTableRowDirective, selector: "[mdbTableRow]", outputs: { rowCreated: "rowCreated", rowRemoved: "rowRemoved" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTableRowDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[mdbTableRow]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { rowCreated: [{
                type: Output
            }], rowRemoved: [{
                type: Output
            }] } });
