// ng-uikit-pro-standard/esm2020/lib/pro/sidenav/sidenav.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SidenavComponent } from './sidenav.component';
import * as i0 from "@angular/core";
export class SidenavModule {
}
SidenavModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SidenavModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SidenavModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SidenavModule, declarations: [SidenavComponent], imports: [CommonModule], exports: [SidenavComponent] });
SidenavModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SidenavModule, imports: [[
            CommonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SidenavModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SidenavComponent,
                    ],
                    exports: [
                        SidenavComponent
                    ],
                    imports: [
                        CommonModule,
                    ]
                }]
        }] });
