// ng-uikit-pro-standard/esm2020/lib/free/badge/mdb-badge.component.mjs
import { coerceBooleanProperty } from '@angular/cdk/coercion';
import { Component, HostBinding, Input, ViewEncapsulation, ChangeDetectionStrategy, } from '@angular/core';
import * as i0 from "@angular/core";
export class MDBBadgeComponent {
    constructor(_el, _renderer) {
        this._el = _el;
        this._renderer = _renderer;
        this._default = false;
        this._primary = false;
        this._secondary = false;
        this._success = false;
        this._info = false;
        this._warning = false;
        this._danger = false;
        this._pill = false;
    }
    get default() {
        return this._default;
    }
    set default(value) {
        this._default = coerceBooleanProperty(value);
    }
    get primary() {
        return this._primary;
    }
    set primary(value) {
        this._primary = coerceBooleanProperty(value);
    }
    get secondary() {
        return this._secondary;
    }
    set secondary(value) {
        this._secondary = coerceBooleanProperty(value);
    }
    get success() {
        return this._success;
    }
    set success(value) {
        this._success = coerceBooleanProperty(value);
    }
    get info() {
        return this._info;
    }
    set info(value) {
        this._info = coerceBooleanProperty(value);
    }
    get warning() {
        return this._warning;
    }
    set warning(value) {
        this._warning = coerceBooleanProperty(value);
    }
    get danger() {
        return this._danger;
    }
    set danger(value) {
        this._danger = coerceBooleanProperty(value);
    }
    get pill() {
        return this._pill;
    }
    set pill(value) {
        this._pill = coerceBooleanProperty(value);
    }
    ngOnInit() {
        this._renderer.addClass(this._el.nativeElement, 'badge');
        if (this.color) {
            const customClassArr = this.color.split(' ');
            customClassArr.forEach((el) => {
                this._renderer.addClass(this._el.nativeElement, el);
            });
        }
    }
}
MDBBadgeComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBadgeComponent, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component });
MDBBadgeComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.2.7", type: MDBBadgeComponent, selector: "mdb-badge", inputs: { default: "default", primary: "primary", secondary: "secondary", success: "success", info: "info", warning: "warning", danger: "danger", pill: "pill", classInside: "classInside", color: "color", class: "class" }, host: { properties: { "class.badge-default": "this.default", "class.badge-primary": "this.primary", "class.badge-secondary": "this.secondary", "class.badge-success": "this.success", "class.badge-info": "this.info", "class.badge-warning": "this.warning", "class.badge-danger": "this.danger", "class.badge-pill": "this.pill" } }, ngImport: i0, template: "<span class=\"{{class}} {{classInside}}\">\n  <ng-content></ng-content>\n</span>\n", styles: [".badge{box-shadow:0 2px 5px #00000029,0 2px 10px #0000001f;border-radius:.125rem;color:#fff!important}.badge-pill{border-radius:10rem;padding-right:.6rem;padding-left:.6rem}.badge-primary{background-color:#4285f4!important;color:#fff!important}.badge-danger{background-color:#ff3547!important;color:#fff!important}.badge-warning{background-color:#fb3!important;color:#fff!important}.badge-success{background-color:#00c851!important;color:#fff!important}.badge-info{background-color:#33b5e5!important;color:#fff!important}.badge-default{background-color:#2bbbad!important;color:#fff!important}.badge-secondary{background-color:#a6c!important;color:#fff!important}.badge-dark{background-color:#212121!important;color:#fff!important}.badge-light{background-color:#e0e0e0!important;color:#000!important}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush, encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBadgeComponent, decorators: [{
            type: Component,
            args: [{ selector: 'mdb-badge', encapsulation: ViewEncapsulation.None, changeDetection: ChangeDetectionStrategy.OnPush, template: "<span class=\"{{class}} {{classInside}}\">\n  <ng-content></ng-content>\n</span>\n", styles: [".badge{box-shadow:0 2px 5px #00000029,0 2px 10px #0000001f;border-radius:.125rem;color:#fff!important}.badge-pill{border-radius:10rem;padding-right:.6rem;padding-left:.6rem}.badge-primary{background-color:#4285f4!important;color:#fff!important}.badge-danger{background-color:#ff3547!important;color:#fff!important}.badge-warning{background-color:#fb3!important;color:#fff!important}.badge-success{background-color:#00c851!important;color:#fff!important}.badge-info{background-color:#33b5e5!important;color:#fff!important}.badge-default{background-color:#2bbbad!important;color:#fff!important}.badge-secondary{background-color:#a6c!important;color:#fff!important}.badge-dark{background-color:#212121!important;color:#fff!important}.badge-light{background-color:#e0e0e0!important;color:#000!important}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }]; }, propDecorators: { default: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-default']
            }], primary: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-primary']
            }], secondary: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-secondary']
            }], success: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-success']
            }], info: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-info']
            }], warning: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-warning']
            }], danger: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-danger']
            }], pill: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.badge-pill']
            }], classInside: [{
                type: Input
            }], color: [{
                type: Input
            }], class: [{
                type: Input
            }] } });
