// ng-uikit-pro-standard/esm2020/lib/pro/mdb-pro.module.mjs
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { AnimatedCardsModule } from './animated-cards/animated-cards.module';
import { FileInputModule } from './file-input/module/mdb-uploader.module';
import { ChipsModule } from './chips/chips.module';
import { ProgressBars } from './progressbars/index';
import { TabsModule } from './tabs-pills/tabset.module';
import { SelectModule } from './material-select/select.module';
import { DatepickerModule } from './date-picker/datepicker.module';
import { TimePickerModule } from './time-picker/timepicker.module';
import { MdbTimePickerModule } from './timepicker/timepicker.module';
import { LightBoxModule } from './lightbox/light-box.module';
import { SidenavModule } from './sidenav/sidenav.module';
import { ChartSimpleModule } from './easy-charts/chart-simple.module';
import { AccordionModule } from './accordion/index';
import { StickyContentModule } from './sticky-content/sticky-content.module';
import { SmoothscrollModule } from './smoothscroll/index';
import { CharCounterModule } from './inputs/char-counter.module';
import { ScrollSpyModule } from './scroll-spy/scroll-spy.module';
import { AutoFormatModule } from './auto-format/auto-format.module';
import { RangeModule } from './range/range.module';
import { AutoCompleterModule } from './auto-completer/auto-completer.module';
import { StepperModule } from './stepper/stepper.module';
import { MdbTreeModule } from './tree-view/tree-view.module';
import { MdbSelectModule } from './select/select.module';
import { MdbOptionModule } from './option/option.module';
import * as i0 from "@angular/core";
import * as i1 from "./animated-cards/animated-cards.module";
import * as i2 from "./progressbars/index";
import * as i3 from "./tabs-pills/tabset.module";
import * as i4 from "./smoothscroll/mdb-page-scroll.module";
import * as i5 from "./inputs/char-counter.module";
export { MdbSelectModule, MdbSelectComponent, MdbSelectFilterComponent, MDB_SELECT_FILTER_VALUE_ACCESSOR, } from './select/index';
export { OptionComponent, OptionGroupComponent, SelectAllOptionComponent, MdbOptionModule, } from './option/index';
export { MdbTreeModule, MdbTreeComponent } from './tree-view/index';
export { MdbStepperComponent, MdbStepComponent, StepperModule } from './stepper/index';
export { MdbAutoCompleterComponent, MdbOptionComponent, MdbAutoCompleterDirective, AutoCompleterModule, MdbAutoCompleterOptionDirective, } from './auto-completer/index';
export { RangeModule, MdbRangeInputComponent } from './range/index';
export { AutoFormatModule, MdbDateFormatDirective, MdbCreditCardDirective, MdbCvvDirective, } from './auto-format/index';
export { ScrollSpyModule, ScrollSpyDirective, ScrollSpyWindowDirective, ScrollSpyElementDirective, ScrollSpyLinkDirective, ScrollSpyService, } from './scroll-spy/index';
export { AnimatedCardsModule, CardRotatingComponent, CardRevealComponent, } from './animated-cards/index';
export { ProgressbarComponent, ProgressbarConfigComponent, ProgressbarModule, ProgressBars, ProgressDirective, ProgressSpinnerComponent, BarComponent, } from './progressbars/index';
export { MaterialChipsComponent, ChipsModule } from './chips/index';
export { TabDirective, TabHeadingDirective, TabsetComponent, TabsetConfig, TabsModule, NgTranscludeDirective, } from './tabs-pills/index';
export { MDBSpinningPreloader } from './preloader/preloader.service';
export { SelectModule, Diacritics, Option, OptionList, SELECT_VALUE_ACCESSOR, SelectComponent, SelectDropdownComponent, } from './material-select/index';
export { MDBDatePickerComponent, DatepickerModule, InputAutoFillDirective, MYDP_VALUE_ACCESSOR, UtilService, LocaleService, FocusDirective, } from './date-picker/index';
export { TimePickerModule, ClockPickerComponent } from './time-picker/index';
export { MdbTimePickerModule, MdbTimePickerDirective, MdbTimePickerComponent, MdbTimepickerToggleComponent, MDB_TIMEPICKER_VALUE_ACCESSOR, MdbTimePickerContentComponent, } from './timepicker/index';
export { LightBoxModule, ImageModalComponent } from './lightbox/index';
export { SidenavComponent, SidenavModule } from './sidenav/index';
export { ChartSimpleModule, EasyPieChartComponent, SimpleChartComponent, } from './easy-charts/index';
export { SBItemComponent, SBItemBodyComponent, SBItemHeadComponent, SqueezeBoxComponent, AccordionModule, } from './accordion/index';
export { MdbStickyDirective, StickyContentModule } from './sticky-content/index';
export { SmoothscrollModule, PageScrollDirective, PageScrollConfig, PageScrollInstance, PageScrollService, PageScrollUtilService, EasingLogic, } from './smoothscroll/index';
export { CharCounterDirective, CharCounterModule } from './inputs/index';
export { MDBFileDropDirective, MDBFileSelectDirective, FileInputModule, MDBUploaderService, humanizeBytes, } from './file-input/index';
const MODULES = [
    AnimatedCardsModule,
    FileInputModule,
    ChipsModule,
    // eslint-disable-next-line import/no-deprecated
    ProgressBars,
    TabsModule,
    SelectModule,
    DatepickerModule,
    TimePickerModule,
    MdbTimePickerModule,
    LightBoxModule,
    SidenavModule,
    ChartSimpleModule,
    AccordionModule,
    StickyContentModule,
    SmoothscrollModule,
    CharCounterModule,
    ScrollSpyModule,
    AutoFormatModule,
    RangeModule,
    AutoCompleterModule,
    StepperModule,
    MdbTreeModule,
    MdbSelectModule,
    MdbOptionModule,
];
export class MDBRootModulePro {
}
MDBRootModulePro.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModulePro, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MDBRootModulePro.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModulePro, imports: [i1.AnimatedCardsModule, ChipsModule, i2.PreloadersModule, i3.TabsModule, SelectModule,
        DatepickerModule,
        TimePickerModule,
        MdbTimePickerModule,
        LightBoxModule,
        SidenavModule,
        ChartSimpleModule,
        AccordionModule,
        StickyContentModule, i4.SmoothscrollModule, i5.CharCounterModule, ScrollSpyModule,
        AutoFormatModule,
        RangeModule,
        AutoCompleterModule,
        StepperModule,
        MdbTreeModule,
        MdbSelectModule,
        MdbOptionModule], exports: [AnimatedCardsModule,
        FileInputModule,
        ChipsModule,
        // eslint-disable-next-line import/no-deprecated
        ProgressBars,
        TabsModule,
        SelectModule,
        DatepickerModule,
        TimePickerModule,
        MdbTimePickerModule,
        LightBoxModule,
        SidenavModule,
        ChartSimpleModule,
        AccordionModule,
        StickyContentModule,
        SmoothscrollModule,
        CharCounterModule,
        ScrollSpyModule,
        AutoFormatModule,
        RangeModule,
        AutoCompleterModule,
        StepperModule,
        MdbTreeModule,
        MdbSelectModule,
        MdbOptionModule] });
MDBRootModulePro.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModulePro, providers: [], imports: [[
            AnimatedCardsModule.forRoot(),
            ChipsModule,
            // eslint-disable-next-line import/no-deprecated
            ProgressBars.forRoot(),
            TabsModule.forRoot(),
            SelectModule,
            DatepickerModule,
            TimePickerModule,
            MdbTimePickerModule,
            LightBoxModule,
            SidenavModule,
            ChartSimpleModule,
            AccordionModule,
            StickyContentModule,
            SmoothscrollModule.forRoot(),
            CharCounterModule.forRoot(),
            ScrollSpyModule,
            AutoFormatModule,
            RangeModule,
            AutoCompleterModule,
            StepperModule,
            MdbTreeModule,
            MdbSelectModule,
            MdbOptionModule,
        ], AnimatedCardsModule,
        FileInputModule,
        ChipsModule,
        // eslint-disable-next-line import/no-deprecated
        ProgressBars,
        TabsModule,
        SelectModule,
        DatepickerModule,
        TimePickerModule,
        MdbTimePickerModule,
        LightBoxModule,
        SidenavModule,
        ChartSimpleModule,
        AccordionModule,
        StickyContentModule,
        SmoothscrollModule,
        CharCounterModule,
        ScrollSpyModule,
        AutoFormatModule,
        RangeModule,
        AutoCompleterModule,
        StepperModule,
        MdbTreeModule,
        MdbSelectModule,
        MdbOptionModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModulePro, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AnimatedCardsModule.forRoot(),
                        ChipsModule,
                        // eslint-disable-next-line import/no-deprecated
                        ProgressBars.forRoot(),
                        TabsModule.forRoot(),
                        SelectModule,
                        DatepickerModule,
                        TimePickerModule,
                        MdbTimePickerModule,
                        LightBoxModule,
                        SidenavModule,
                        ChartSimpleModule,
                        AccordionModule,
                        StickyContentModule,
                        SmoothscrollModule.forRoot(),
                        CharCounterModule.forRoot(),
                        ScrollSpyModule,
                        AutoFormatModule,
                        RangeModule,
                        AutoCompleterModule,
                        StepperModule,
                        MdbTreeModule,
                        MdbSelectModule,
                        MdbOptionModule,
                    ],
                    exports: [MODULES],
                    providers: [],
                    schemas: [NO_ERRORS_SCHEMA],
                }]
        }] });
export class MDBBootstrapModulePro {
    static forRoot() {
        return { ngModule: MDBRootModulePro };
    }
}
MDBBootstrapModulePro.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulePro, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MDBBootstrapModulePro.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulePro, exports: [AnimatedCardsModule,
        FileInputModule,
        ChipsModule,
        // eslint-disable-next-line import/no-deprecated
        ProgressBars,
        TabsModule,
        SelectModule,
        DatepickerModule,
        TimePickerModule,
        MdbTimePickerModule,
        LightBoxModule,
        SidenavModule,
        ChartSimpleModule,
        AccordionModule,
        StickyContentModule,
        SmoothscrollModule,
        CharCounterModule,
        ScrollSpyModule,
        AutoFormatModule,
        RangeModule,
        AutoCompleterModule,
        StepperModule,
        MdbTreeModule,
        MdbSelectModule,
        MdbOptionModule] });
MDBBootstrapModulePro.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulePro, imports: [AnimatedCardsModule,
        FileInputModule,
        ChipsModule,
        // eslint-disable-next-line import/no-deprecated
        ProgressBars,
        TabsModule,
        SelectModule,
        DatepickerModule,
        TimePickerModule,
        MdbTimePickerModule,
        LightBoxModule,
        SidenavModule,
        ChartSimpleModule,
        AccordionModule,
        StickyContentModule,
        SmoothscrollModule,
        CharCounterModule,
        ScrollSpyModule,
        AutoFormatModule,
        RangeModule,
        AutoCompleterModule,
        StepperModule,
        MdbTreeModule,
        MdbSelectModule,
        MdbOptionModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulePro, decorators: [{
            type: NgModule,
            args: [{ exports: [MODULES] }]
        }] });
