// ng-uikit-pro-standard/esm2020/lib/pro/smoothscroll/mdb-page-scroll.module.mjs
/**
 * Created by sebastianfuss on 03.09.16.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PageScrollService } from './mdb-page-scroll.service';
import { PageScrollDirective } from './mdb-page-scroll.directive';
import * as i0 from "@angular/core";
export class SmoothscrollModule {
    static forRoot() {
        return {
            ngModule: SmoothscrollModule,
            providers: [{ provide: PageScrollService, useClass: PageScrollService }],
        };
    }
}
SmoothscrollModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SmoothscrollModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SmoothscrollModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SmoothscrollModule, declarations: [PageScrollDirective], imports: [CommonModule], exports: [PageScrollDirective] });
SmoothscrollModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SmoothscrollModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SmoothscrollModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [PageScrollDirective],
                    exports: [PageScrollDirective],
                }]
        }] });
