// ng-uikit-pro-standard/esm2020/lib/pro/lightbox/light-box.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { ImageModalComponent } from './image-popup';
import * as i0 from "@angular/core";
export class LightBoxModule {
}
LightBoxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: LightBoxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LightBoxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: LightBoxModule, declarations: [ImageModalComponent], imports: [CommonModule, FormsModule], exports: [ImageModalComponent] });
LightBoxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: LightBoxModule, imports: [[CommonModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: LightBoxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule],
                    declarations: [ImageModalComponent],
                    exports: [ImageModalComponent]
                }]
        }] });
