// ng-uikit-pro-standard/esm2020/lib/mdb.module.mjs
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { MDBBootstrapModule } from './free/mdb-free.module';
import { MDBBootstrapModulePro } from './pro/mdb-pro.module';
import * as i0 from "@angular/core";
import * as i1 from "./free/mdb-free.module";
import * as i2 from "./pro/mdb-pro.module";
export { MDBBootstrapModule } from './free/mdb-free.module';
export { MDBBootstrapModulePro } from './pro/mdb-pro.module';
const MODULES = [MDBBootstrapModule, MDBBootstrapModulePro];
export class MDBRootModules {
}
MDBRootModules.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModules, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MDBRootModules.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModules, imports: [i1.MDBRootModule, i2.MDBRootModulePro], exports: [MDBBootstrapModule, MDBBootstrapModulePro] });
MDBRootModules.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModules, providers: [], imports: [[MDBBootstrapModule.forRoot(), MDBBootstrapModulePro.forRoot()], MDBBootstrapModule, MDBBootstrapModulePro] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBRootModules, decorators: [{
            type: NgModule,
            args: [{
                    imports: [MDBBootstrapModule.forRoot(), MDBBootstrapModulePro.forRoot()],
                    exports: MODULES,
                    providers: [],
                    schemas: [NO_ERRORS_SCHEMA],
                }]
        }] });
export class MDBBootstrapModulesPro {
    static forRoot() {
        return { ngModule: MDBRootModules };
    }
}
MDBBootstrapModulesPro.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulesPro, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MDBBootstrapModulesPro.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulesPro, exports: [MDBBootstrapModule, MDBBootstrapModulePro] });
MDBBootstrapModulesPro.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulesPro, imports: [MDBBootstrapModule, MDBBootstrapModulePro] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MDBBootstrapModulesPro, decorators: [{
            type: NgModule,
            args: [{ exports: MODULES }]
        }] });
