// ng-uikit-pro-standard/esm2020/lib/pro/easy-charts/chart-simple.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SimpleChartComponent } from './chart-simple.component';
import { EasyPieChartComponent } from './chart-smallpie.component';
import * as i0 from "@angular/core";
export class ChartSimpleModule {
}
ChartSimpleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ChartSimpleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ChartSimpleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ChartSimpleModule, declarations: [SimpleChartComponent, EasyPieChartComponent], imports: [CommonModule], exports: [SimpleChartComponent, EasyPieChartComponent] });
ChartSimpleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ChartSimpleModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ChartSimpleModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SimpleChartComponent, EasyPieChartComponent],
                    exports: [SimpleChartComponent, EasyPieChartComponent],
                    imports: [CommonModule],
                }]
        }] });
