// ng-uikit-pro-standard/esm2020/lib/pro/timepicker/timepicker.module.mjs
import { OverlayModule } from '@angular/cdk/overlay';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { A11yModule } from '@angular/cdk/a11y';
import { ButtonsModule } from '../../free/buttons/buttons.module';
import { WavesModule } from '../../free/waves/waves.module';
import { MdbTimepickerToggleComponent } from './timepicker-toggle.component';
import { MdbTimePickerDirective } from './timepicker.directive';
import { MdbTimePickerComponent } from './timepicker.component';
import { MdbTimePickerContentComponent } from './timepicker.content';
import * as i0 from "@angular/core";
import * as i1 from "../../free/buttons/buttons.module";
import * as i2 from "../../free/waves/waves.module";
export class MdbTimePickerModule {
}
MdbTimePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTimePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdbTimePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTimePickerModule, bootstrap: [MdbTimePickerContentComponent], declarations: [MdbTimePickerComponent,
        MdbTimepickerToggleComponent,
        MdbTimePickerDirective,
        MdbTimePickerContentComponent], imports: [CommonModule,
        OverlayModule,
        A11yModule, i1.ButtonsModule, i2.WavesModule], exports: [MdbTimePickerComponent, MdbTimepickerToggleComponent, MdbTimePickerDirective] });
MdbTimePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTimePickerModule, imports: [[
            CommonModule,
            OverlayModule,
            A11yModule,
            ButtonsModule.forRoot(),
            WavesModule.forRoot(),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTimePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        OverlayModule,
                        A11yModule,
                        ButtonsModule.forRoot(),
                        WavesModule.forRoot(),
                    ],
                    declarations: [
                        MdbTimePickerComponent,
                        MdbTimepickerToggleComponent,
                        MdbTimePickerDirective,
                        MdbTimePickerContentComponent,
                    ],
                    exports: [MdbTimePickerComponent, MdbTimepickerToggleComponent, MdbTimePickerDirective],
                    bootstrap: [MdbTimePickerContentComponent],
                    entryComponents: [MdbTimePickerContentComponent],
                }]
        }] });
