// ng-uikit-pro-standard/esm2020/lib/pro/time-picker/timepicker.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { ClockPickerComponent } from './timepicker.component';
import { ButtonsModule } from '../../free/buttons/buttons.module';
import { WavesModule } from '../../free/waves/waves.module';
import * as i0 from "@angular/core";
import * as i1 from "../../free/buttons/buttons.module";
import * as i2 from "../../free/waves/waves.module";
export class TimePickerModule {
}
TimePickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: TimePickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TimePickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: TimePickerModule, declarations: [ClockPickerComponent], imports: [CommonModule, FormsModule, i1.ButtonsModule, i2.WavesModule], exports: [ClockPickerComponent] });
TimePickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: TimePickerModule, imports: [[CommonModule, FormsModule, ButtonsModule.forRoot(), WavesModule.forRoot()]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: TimePickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule, ButtonsModule.forRoot(), WavesModule.forRoot()],
                    declarations: [ClockPickerComponent],
                    exports: [ClockPickerComponent],
                }]
        }] });
