// ng-uikit-pro-standard/esm2020/lib/pro/progressbars/index.mjs
import { NgModule } from '@angular/core';
import { ProgressbarModule } from './progressbar.module';
import { MdProgressSpinnerModule } from './progress-spinner-module/index';
import { MdProgressBarModule } from './progress-bars-module/index';
import * as i0 from "@angular/core";
import * as i1 from "./progress-bars-module/index";
import * as i2 from "./progress-spinner-module/index";
import * as i3 from "./progressbar.module";
export { BarComponent } from './bar.component';
export { ProgressDirective } from './progress.directive';
export { ProgressbarComponent } from './progressbar.component';
export { ProgressbarModule } from './progressbar.module';
export { ProgressbarConfigComponent } from './progressbar.config.component';
export { ProgressSpinnerComponent } from './progress-spinner.component';
export * from './progress-bars-module/index';
export * from './progress-spinner-module/index';
const MATERIAL_MODULES = [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule];
export class PreloadersModule {
}
PreloadersModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: PreloadersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PreloadersModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: PreloadersModule, imports: [i1.MdProgressBarModule, i2.MdProgressSpinnerModule, i3.ProgressbarModule], exports: [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] });
PreloadersModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: PreloadersModule, imports: [[
            MdProgressBarModule.forRoot(),
            MdProgressSpinnerModule.forRoot(),
            ProgressbarModule.forRoot(),
        ], MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: PreloadersModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        MdProgressBarModule.forRoot(),
                        MdProgressSpinnerModule.forRoot(),
                        ProgressbarModule.forRoot(),
                    ],
                    exports: MATERIAL_MODULES,
                }]
        }] });
export class ProgressBars {
    static forRoot() {
        return { ngModule: PreloadersModule };
    }
}
ProgressBars.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ProgressBars, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProgressBars.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ProgressBars, imports: [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule], exports: [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] });
ProgressBars.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ProgressBars, imports: [MATERIAL_MODULES, MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ProgressBars, decorators: [{
            type: NgModule,
            args: [{
                    imports: MATERIAL_MODULES,
                    exports: MATERIAL_MODULES,
                }]
        }] });
