// ng-uikit-pro-standard/esm2020/lib/pro/progressbars/progress.directive.mjs
import { coerceBooleanProperty, coerceNumberProperty, } from '@angular/cdk/coercion';
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class ProgressDirective {
    constructor() {
        this._animate = false;
        this.addClass = true;
        this.bars = [];
        this._max = 100;
    }
    /** if `true` changing value of progress bar will be animated (note: not supported by Bootstrap 4) */
    get animate() {
        return this._animate;
    }
    set animate(value) {
        this._animate = coerceBooleanProperty(value);
    }
    /** maximum total value of progress element */
    get max() {
        return this._max;
    }
    set max(value) {
        const v = coerceNumberProperty(value);
        this._max = v;
        this.bars.forEach((bar) => {
            bar.recalculatePercentage();
        });
    }
    addBar(bar) {
        if (!this.animate) {
            bar.transition = 'none';
        }
        this.bars.push(bar);
    }
    removeBar(bar) {
        this.bars.splice(this.bars.indexOf(bar), 1);
    }
}
ProgressDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ProgressDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
ProgressDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.7", type: ProgressDirective, selector: "mdbProgress, [mdbProgress]", inputs: { animate: "animate", max: "max" }, host: { properties: { "attr.max": "this.max", "class.progress": "this.addClass" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: ProgressDirective, decorators: [{
            type: Directive,
            args: [{ selector: 'mdbProgress, [mdbProgress]' }]
        }], propDecorators: { animate: [{
                type: Input
            }], max: [{
                type: HostBinding,
                args: ['attr.max']
            }, {
                type: Input
            }], addClass: [{
                type: HostBinding,
                args: ['class.progress']
            }] } });
