// ng-uikit-pro-standard/esm2020/lib/pro/animated-cards/animated-cards.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardRevealComponent } from './card-reveal.component';
import { CardRotatingComponent } from './card-rotating.component';
import * as i0 from "@angular/core";
export class AnimatedCardsModule {
    static forRoot() {
        return { ngModule: AnimatedCardsModule, providers: [] };
    }
}
AnimatedCardsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AnimatedCardsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AnimatedCardsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AnimatedCardsModule, declarations: [CardRevealComponent, CardRotatingComponent], imports: [CommonModule], exports: [CardRevealComponent, CardRotatingComponent] });
AnimatedCardsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AnimatedCardsModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AnimatedCardsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [CardRevealComponent, CardRotatingComponent],
                    exports: [CardRevealComponent, CardRotatingComponent],
                }]
        }] });
