// ng-uikit-pro-standard/esm2020/lib/free/navbars/navlinks.component.mjs
import { Component, ContentChildren, ElementRef, EventEmitter, Output, } from '@angular/core';
import { RouterLinkWithHref } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "./navbar.service";
export class NavlinksComponent {
    constructor(_navbarService, renderer) {
        this._navbarService = _navbarService;
        this.renderer = renderer;
        this.linkClick = new EventEmitter();
    }
    ngAfterContentInit() {
        setTimeout(() => {
            this.links.forEach((link) => {
                this.renderer.listen(link.nativeElement, 'click', () => {
                    this._navbarService.setNavbarLinkClicks();
                });
            });
        }, 0);
    }
}
NavlinksComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: NavlinksComponent, deps: [{ token: i1.NavbarService }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component });
NavlinksComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.2.7", type: NavlinksComponent, selector: "navlinks", outputs: { linkClick: "linkClick" }, queries: [{ propertyName: "links", predicate: RouterLinkWithHref, descendants: true, read: ElementRef }], ngImport: i0, template: `
    <ng-content></ng-content>
  `, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: NavlinksComponent, decorators: [{
            type: Component,
            args: [{
                    // eslint-disable-next-line @angular-eslint/component-selector
                    selector: 'navlinks',
                    template: `
    <ng-content></ng-content>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.NavbarService }, { type: i0.Renderer2 }]; }, propDecorators: { links: [{
                type: ContentChildren,
                args: [RouterLinkWithHref, { read: ElementRef, descendants: true }]
            }], linkClick: [{
                type: Output
            }] } });
