// ng-uikit-pro-standard/esm2020/lib/pro/stepper/stepper.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbStepComponent } from './step.component';
import { MdbStepperComponent } from './stepper.component';
import * as i0 from "@angular/core";
export class StepperModule {
}
StepperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StepperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StepperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StepperModule, declarations: [MdbStepperComponent,
        MdbStepComponent], imports: [CommonModule], exports: [MdbStepperComponent,
        MdbStepComponent] });
StepperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StepperModule, imports: [[
            CommonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StepperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        MdbStepperComponent,
                        MdbStepComponent
                    ],
                    exports: [
                        MdbStepperComponent,
                        MdbStepComponent
                    ],
                    imports: [
                        CommonModule,
                    ]
                }]
        }] });
