// ng-uikit-pro-standard/esm2020/lib/free/waves/waves.module.mjs
import { NgModule } from '@angular/core';
import { WavesDirective } from './waves-effect.directive';
import * as i0 from "@angular/core";
export class WavesModule {
    static forRoot() {
        return { ngModule: WavesModule, providers: [] };
    }
}
WavesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: WavesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
WavesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: WavesModule, declarations: [WavesDirective], exports: [WavesDirective] });
WavesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: WavesModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: WavesModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [WavesDirective],
                    exports: [WavesDirective],
                }]
        }] });
