// ng-uikit-pro-standard/esm2020/lib/pro/inputs/char-counter.module.mjs
import { NgModule } from '@angular/core';
import { CharCounterDirective } from './char-counter.directive';
import * as i0 from "@angular/core";
export class CharCounterModule {
    static forRoot() {
        return { ngModule: CharCounterModule, providers: [] };
    }
}
CharCounterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: CharCounterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CharCounterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: CharCounterModule, declarations: [CharCounterDirective], exports: [CharCounterDirective] });
CharCounterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: CharCounterModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: CharCounterModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CharCounterDirective],
                    exports: [CharCounterDirective],
                }]
        }] });
