// ng-uikit-pro-standard/esm2020/lib/pro/inputs/char-counter.directive.mjs
import { coerceNumberProperty } from '@angular/cdk/coercion';
import { Directive, Input, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
export class CharCounterDirective {
    constructor(_elRef, _renderer) {
        this._elRef = _elRef;
        this._renderer = _renderer;
        this._length = 20;
    }
    get length() {
        return this._length;
    }
    set length(value) {
        this._length = coerceNumberProperty(value);
    }
    ngOnInit() {
        // Inititalise a new <span> element for the count display and render it below the host component.
        this.textContainer = this._renderer.createElement('p');
        this._renderer.appendChild(this._elRef.nativeElement.parentElement, this.textContainer);
        this._renderer.addClass(this.textContainer, 'chars');
        this.textContainer.innerHTML = '0/' + this.length;
        this._renderer.setStyle(this.textContainer, 'display', 'none');
    }
    onKeyUp() {
        this.textContainer.innerHTML = this._elRef.nativeElement.value.length + '/' + this.length;
        if (this._elRef.nativeElement.value.length > this.length) {
            this._renderer.addClass(this._elRef.nativeElement, 'invalid');
        }
        else {
            this._renderer.removeClass(this._elRef.nativeElement, 'invalid');
        }
    }
    hide() {
        this._renderer.setStyle(this.textContainer, 'display', 'none');
    }
    show() {
        this.onKeyUp();
        this._renderer.setStyle(this.textContainer, 'display', 'block');
    }
}
CharCounterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: CharCounterDirective, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive });
CharCounterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.2.7", type: CharCounterDirective, selector: "[mdbCharCounter]", inputs: { length: "length" }, host: { listeners: { "input": "onKeyUp()", "blur": "hide()", "focus": "show()" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: CharCounterDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[mdbCharCounter]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }]; }, propDecorators: { length: [{
                type: Input
            }], onKeyUp: [{
                type: HostListener,
                args: ['input']
            }], hide: [{
                type: HostListener,
                args: ['blur']
            }], show: [{
                type: HostListener,
                args: ['focus']
            }] } });
