// ng-uikit-pro-standard/esm2020/lib/pro/sticky-content/sticky-content.module.mjs
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { MdbStickyDirective } from './sticky-content.directive';
import * as i0 from "@angular/core";
export { MdbStickyDirective };
export class StickyContentModule {
}
StickyContentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyContentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StickyContentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyContentModule, declarations: [MdbStickyDirective], imports: [CommonModule, FormsModule], exports: [MdbStickyDirective] });
StickyContentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyContentModule, imports: [[CommonModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyContentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, FormsModule],
                    declarations: [MdbStickyDirective],
                    exports: [MdbStickyDirective]
                }]
        }] });
