// ng-uikit-pro-standard/esm2020/lib/pro/tree-view/tree-view.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbTreeComponent } from './tree-view.component';
import { CheckboxModule } from '../../free/checkbox/checkbox.module';
import * as i0 from "@angular/core";
export class MdbTreeModule {
}
MdbTreeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTreeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MdbTreeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTreeModule, declarations: [MdbTreeComponent], imports: [CommonModule, CheckboxModule], exports: [MdbTreeComponent] });
MdbTreeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTreeModule, imports: [[CommonModule, CheckboxModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: MdbTreeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CheckboxModule],
                    declarations: [MdbTreeComponent],
                    exports: [MdbTreeComponent],
                }]
        }] });
