// ng-uikit-pro-standard/esm2020/lib/pro/auto-completer/auto-completer.module.mjs
import { NgModule } from '@angular/core';
import { MdbAutoCompleterComponent } from './components/mdb-auto-completer.component';
import { MdbOptionComponent } from './components/mdb-option.component';
import { MdbAutoCompleterDirective } from './directives/mdb-auto-completer.directive';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';
import { MdbAutoCompleterOptionDirective } from './directives/mdb-auto-completer-option.directive';
import * as i0 from "@angular/core";
export class AutoCompleterModule {
}
AutoCompleterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AutoCompleterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AutoCompleterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AutoCompleterModule, declarations: [MdbAutoCompleterComponent,
        MdbOptionComponent,
        MdbAutoCompleterDirective,
        MdbAutoCompleterOptionDirective], imports: [CommonModule, HttpClientModule, FormsModule], exports: [MdbAutoCompleterComponent,
        MdbOptionComponent,
        MdbAutoCompleterDirective,
        MdbAutoCompleterOptionDirective] });
AutoCompleterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AutoCompleterModule, imports: [[CommonModule, HttpClientModule, FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: AutoCompleterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, HttpClientModule, FormsModule],
                    declarations: [
                        MdbAutoCompleterComponent,
                        MdbOptionComponent,
                        MdbAutoCompleterDirective,
                        MdbAutoCompleterOptionDirective,
                    ],
                    exports: [
                        MdbAutoCompleterComponent,
                        MdbOptionComponent,
                        MdbAutoCompleterDirective,
                        MdbAutoCompleterOptionDirective,
                    ],
                }]
        }] });
