// ng-uikit-pro-standard/esm2020/lib/free/breadcrumbs/breadcrumb.module.mjs
import { CommonModule } from '@angular/common';
import { MdbBreadcrumbComponent } from './mdb-breadcrumb.component';
import { MdbBreadcrumbItemComponent } from './mdb-breadcrumb-item.component';
import { NgModule } from '@angular/core';
import * as i0 from "@angular/core";
export class BreadcrumbModule {
}
BreadcrumbModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: BreadcrumbModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BreadcrumbModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: BreadcrumbModule, declarations: [MdbBreadcrumbComponent, MdbBreadcrumbItemComponent], imports: [CommonModule], exports: [MdbBreadcrumbComponent, MdbBreadcrumbItemComponent] });
BreadcrumbModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: BreadcrumbModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: BreadcrumbModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [MdbBreadcrumbComponent, MdbBreadcrumbItemComponent],
                    exports: [MdbBreadcrumbComponent, MdbBreadcrumbItemComponent]
                }]
        }] });
