// ng-uikit-pro-standard/esm2020/lib/free/carousel/slide.component.mjs
import { Component, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SlideComponent {
    constructor(el) {
        this.animated = false;
        this.directionNext = false;
        this.directionLeft = false;
        this.directionPrev = false;
        this.directionRight = false;
        /** Wraps element by appropriate CSS classes */
        this.el = null;
        this.el = el;
    }
}
SlideComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SlideComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
SlideComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.2.7", type: SlideComponent, selector: "mdb-slide, mdb-carousel-item", inputs: { active: "active" }, host: { properties: { "class.active": "this.active", "class.animated": "this.animated", "class.carousel-item-next": "this.directionNext", "class.carousel-item-left": "this.directionLeft", "class.carousel-item-prev": "this.directionPrev", "class.carousel-item-right": "this.directionRight", "class.carousel-item": "this.el" } }, ngImport: i0, template: `
    <ng-content></ng-content>
  `, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: SlideComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'mdb-slide, mdb-carousel-item',
                    template: `
    <ng-content></ng-content>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { active: [{
                type: HostBinding,
                args: ['class.active']
            }, {
                type: Input
            }], animated: [{
                type: HostBinding,
                args: ['class.animated']
            }], directionNext: [{
                type: HostBinding,
                args: ['class.carousel-item-next']
            }], directionLeft: [{
                type: HostBinding,
                args: ['class.carousel-item-left']
            }], directionPrev: [{
                type: HostBinding,
                args: ['class.carousel-item-prev']
            }], directionRight: [{
                type: HostBinding,
                args: ['class.carousel-item-right']
            }], el: [{
                type: HostBinding,
                args: ['class.carousel-item']
            }] } });
