// ng-uikit-pro-standard/esm2020/lib/free/sticky-header/sticky-header.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StickyHeaderDirective } from './sticky-header.directive';
import * as i0 from "@angular/core";
export class StickyHeaderModule {
}
StickyHeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyHeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
StickyHeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyHeaderModule, declarations: [StickyHeaderDirective], imports: [CommonModule], exports: [StickyHeaderDirective] });
StickyHeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyHeaderModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: StickyHeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [StickyHeaderDirective],
                    exports: [StickyHeaderDirective],
                    imports: [CommonModule],
                }]
        }] });
