// ng-uikit-pro-standard/esm2020/lib/free/input-utilities/input-utilities.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbErrorDirective } from './error.directive';
import { MdbSuccessDirective } from './success.directive';
import { MdbValidateDirective } from './validate.directive';
import * as i0 from "@angular/core";
export class InputUtilitiesModule {
}
InputUtilitiesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: InputUtilitiesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputUtilitiesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: InputUtilitiesModule, declarations: [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective], imports: [CommonModule], exports: [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective] });
InputUtilitiesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: InputUtilitiesModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.7", ngImport: i0, type: InputUtilitiesModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective],
                    exports: [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective],
                }]
        }] });
